src/store/tools/rulers.ts

```diff
@@ -62,105 +49,22 @@ export const useRulerStore = defineStore('ruler', () => {
     }, {});
   });
 
-  // --- actions --- //
-
-  function addRuler(this: _This, ruler: RulerPatch) {
-    const id = this.$id.nextID();
-    const color = ruler.color ?? getNextColor();
-    set(rulerByID.value, id, createRulerWithDefaults({ ...ruler, id, color }));
-    rulerIDs.value.push(id);
-    return id;
-  }
-
-  function removeRuler(id: string) {
-    if (id in rulerByID.value) {
-      removeFromArray(rulerIDs.value, id);
-      del(rulerByID.value, id);
-    }
-  }
-
-  function updateRuler(id: string, patch: RulerPatch) {
-    if (id in rulerByID.value) {
-      set(rulerByID.value, id, { ...rulerByID.value[id], ...patch, id });
-    }
-  }
-
-  function jumpToRuler(rulerID: string) {
-    const ruler = rulerByID.value[rulerID];
-    const { currentImageID, currentImageMetadata } = useCurrentImage();
-
-    const imageID = currentImageID.value;
-    if (!imageID || ruler.imageID !== imageID) return;
-
-    const rulerImageFrame = frameOfReferenceToImageSliceAndAxis(
-      ruler.frameOfReference,
-      currentImageMetadata.value
-    );
-
-    if (!rulerImageFrame) return;
-
-    const viewStore = useViewStore();
-    const relevantViewIDs = viewStore.viewIDs.filter((viewID) => {
-      const viewSpec = viewStore.viewSpecs[viewID];
-      const viewDir = viewSpec.props.viewDirection as LPSAxisDir | undefined;
-      return viewDir && getLPSAxisFromDir(viewDir) === rulerImageFrame.axis;
-    });
-
-    const viewConfigStore = useViewConfigStore();
-    relevantViewIDs.forEach((viewID) => {
-      viewConfigStore.updateSliceConfig(viewID, imageID, {
-        slice: ruler.slice!,
-      });
-    });
-  }
-
-  // --- tool activation --- //
-
-  function activateTool(this: _This) {
-    return true;
-  }
-
-  function deactivateTool() {}
-
   // --- serialization --- //
 
   function serialize(state: StateFile) {
-    state.manifest.tools.rulers = rulers.value
-      .filter((ruler) => !ruler.placing)
-      // If parent image is DICOM, save VolumeKey
-      .map(({ imageID, ...rest }) => ({
-        imageID: getDataID(imageID),
-        ...rest,
-      }));
+    state.manifest.tools.rulers = serializeTools();
   }
 
-  function deserialize(
-    this: _This,
-    manifest: Manifest,
-    dataIDMap: Record<string, string>
-  ) {
-    const rulersInState = manifest.tools.rulers;
-
-    rulersInState
-      .map(({ imageID, ...rest }) => {
-        const newID = dataIDMap[imageID];
-        return {
-          ...rest,
-          imageID: findImageID(newID),
-        };
-      })
-      .forEach((ruler) => {
-        addRuler.call(this, ruler);
-      });
+  function deserialize(manifest: Manifest, dataIDMap: Record<string, string>) {
+    deserializeTools(manifest.tools.rulers, dataIDMap);
   }
 
   return {
-    rulers,
+    ...toolStoreProps,
```

Comment:
Do we still need `toolStoreProps` spread here? Is there anything that is used?

Reply:
Oh, I see `activateTool` and `deactivateTool` now. Let's just assign them, since we are polluting the exposed API here with stuff like `addTool`, which is aliased as `addRuler`.

Reply:
Another half baked thing there.  Would be nice to have a polymorphic toolStore shape no?  Makes Measurement.vue potentially nice dealing with single collection of tools (but the branded RectangleID type is sort of messing it up at the moment).  

So I will bite the bullet now and change all uses of `rulerStore.addRuler/updateRuler/etc` to `rulerStore.addTool/updateTool/etcTool`

Reply:
I actually changed `MeasurementList` in the vue3 branch to be specific to the tool, since not all tools report the same info. So I have a `MeasurementRulerList` right now.

Reply:
Let's just keep `addRuler/etc` for now. A general toolStore shape would be nice, but I want to be cautious in jumping into that first before we've had a chance to add more tools that aren't necessarily super similar to each other.

Reply:
OK, I'll assign `activateTool` and `deactivateTool`, not spread the wrapped props and adjust `MeasurementList`